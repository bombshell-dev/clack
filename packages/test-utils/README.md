# `@clack/test-utils`

A bunch of useful test utiltiies for use inside clack.
