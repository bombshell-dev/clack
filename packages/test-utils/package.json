{
  "name": "@clack/test-utils",
  "private": true,
  "version": "1.0.0-alpha.1",
  "type": "module",
  "main": "./dist/index.js",
  "module": "./dist/index.js",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "./package.json": "./package.json"
  },
  "types": "./dist/index.d.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bombshell-dev/clack.git",
    "directory": "packages/test-utils"
  },
  "bugs": {
    "url": "https://github.com/bombshell-dev/clack/issues"
  },
  "homepage": "https://github.com/bombshell-dev/clack/tree/main/packages/test-utils#readme",
  "files": ["dist", "CHANGELOG.md"],
  "author": {
    "name": "James Garbutt",
    "url": "https://github.com/43081j"
  },
  "license": "MIT",
  "packageManager": "pnpm@9.14.2",
  "scripts": {
    "build": "tsc",
    "prepack": "pnpm build"
  },
  "dependencies": {
  },
  "devDependencies": {
  }
}
