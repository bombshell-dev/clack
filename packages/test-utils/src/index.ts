export { MockReadable } from './mock-readable.js';
export { MockWritable } from './mock-writable.js';

export function nextTick(): Promise<void> {
	return new Promise<void>((resolve) => {
		setTimeout(resolve, 0);
	});
}
