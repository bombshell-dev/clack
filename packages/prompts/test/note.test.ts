import colors from 'picocolors';
import { afterAll, afterEach, beforeAll, beforeEach, describe, expect, test, vi } from 'vitest';
import * as prompts from '../src/index.js';
import { MockReadable, MockWritable } from './test-utils.js';

describe.each(['true', 'false'])('note (isCI = %s)', (isCI) => {
	let originalCI: string | undefined;
	let output: MockWritable;
	let input: MockReadable;

	beforeAll(() => {
		originalCI = process.env.CI;
		process.env.CI = isCI;
	});

	afterAll(() => {
		process.env.CI = originalCI;
	});

	beforeEach(() => {
		output = new MockWritable();
		input = new MockReadable();
	});

	afterEach(() => {
		vi.restoreAllMocks();
	});

	test('renders message with title', () => {
		prompts.note('message', 'title', {
			input,
			output,
		});

		expect(output.buffer).toMatchSnapshot();
	});

	test('renders as wide as longest line', () => {
		prompts.note('short\nsomewhat questionably long line', 'title', {
			input,
			output,
		});

		expect(output.buffer).toMatchSnapshot();
	});

	test('formatter which adds length works', () => {
		prompts.note('line 0\nline 1\nline 2', 'title', {
			format: (line) => `* ${line} *`,
			input,
			output,
		});

		expect(output.buffer).toMatchSnapshot();
	});

	test('formatter which adds colors works', () => {
		prompts.note('line 0\nline 1\nline 2', 'title', {
			format: (line) => colors.red(line),
			input,
			output,
		});

		expect(output.buffer).toMatchSnapshot();
	});
});
