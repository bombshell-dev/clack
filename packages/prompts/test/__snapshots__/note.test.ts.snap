// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`note (isCI = false) > formatter which adds colors works 1`] = `
[
  "[90mâ”‚[39m
[32mâ—‡[39m  [0mtitle[0m [90mâ”€â”€â•®[39m
[90mâ”‚[39m          [90mâ”‚[39m
[90mâ”‚[39m  [31mline 0[39m  [90mâ”‚[39m
[90mâ”‚[39m  [31mline 1[39m  [90mâ”‚[39m
[90mâ”‚[39m  [31mline 2[39m  [90mâ”‚[39m
[90mâ”‚[39m          [90mâ”‚[39m
[90mâ”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯[39m
",
]
`;

exports[`note (isCI = false) > formatter which adds length works 1`] = `
[
  "[90mâ”‚[39m
[32mâ—‡[39m  [0mtitle[0m [90mâ”€â”€â”€â”€â”€â”€â•®[39m
[90mâ”‚[39m              [90mâ”‚[39m
[90mâ”‚[39m  * line 0 *  [90mâ”‚[39m
[90mâ”‚[39m  * line 1 *  [90mâ”‚[39m
[90mâ”‚[39m  * line 2 *  [90mâ”‚[39m
[90mâ”‚[39m              [90mâ”‚[39m
[90mâ”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯[39m
",
]
`;

exports[`note (isCI = false) > renders as wide as longest line 1`] = `
[
  "[90mâ”‚[39m
[32mâ—‡[39m  [0mtitle[0m [90mâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®[39m
[90mâ”‚[39m                                   [90mâ”‚[39m
[90mâ”‚[39m  [2mshort[22m                            [90mâ”‚[39m
[90mâ”‚[39m  [2msomewhat questionably long line[22m  [90mâ”‚[39m
[90mâ”‚[39m                                   [90mâ”‚[39m
[90mâ”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯[39m
",
]
`;

exports[`note (isCI = false) > renders message with title 1`] = `
[
  "[90mâ”‚[39m
[32mâ—‡[39m  [0mtitle[0m [90mâ”€â”€â”€â•®[39m
[90mâ”‚[39m           [90mâ”‚[39m
[90mâ”‚[39m  [2mmessage[22m  [90mâ”‚[39m
[90mâ”‚[39m           [90mâ”‚[39m
[90mâ”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯[39m
",
]
`;

exports[`note (isCI = true) > formatter which adds colors works 1`] = `
[
  "[90mâ”‚[39m
[32mâ—‡[39m  [0mtitle[0m [90mâ”€â”€â•®[39m
[90mâ”‚[39m          [90mâ”‚[39m
[90mâ”‚[39m  [31mline 0[39m  [90mâ”‚[39m
[90mâ”‚[39m  [31mline 1[39m  [90mâ”‚[39m
[90mâ”‚[39m  [31mline 2[39m  [90mâ”‚[39m
[90mâ”‚[39m          [90mâ”‚[39m
[90mâ”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯[39m
",
]
`;

exports[`note (isCI = true) > formatter which adds length works 1`] = `
[
  "[90mâ”‚[39m
[32mâ—‡[39m  [0mtitle[0m [90mâ”€â”€â”€â”€â”€â”€â•®[39m
[90mâ”‚[39m              [90mâ”‚[39m
[90mâ”‚[39m  * line 0 *  [90mâ”‚[39m
[90mâ”‚[39m  * line 1 *  [90mâ”‚[39m
[90mâ”‚[39m  * line 2 *  [90mâ”‚[39m
[90mâ”‚[39m              [90mâ”‚[39m
[90mâ”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯[39m
",
]
`;

exports[`note (isCI = true) > renders as wide as longest line 1`] = `
[
  "[90mâ”‚[39m
[32mâ—‡[39m  [0mtitle[0m [90mâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®[39m
[90mâ”‚[39m                                   [90mâ”‚[39m
[90mâ”‚[39m  [2mshort[22m                            [90mâ”‚[39m
[90mâ”‚[39m  [2msomewhat questionably long line[22m  [90mâ”‚[39m
[90mâ”‚[39m                                   [90mâ”‚[39m
[90mâ”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯[39m
",
]
`;

exports[`note (isCI = true) > renders message with title 1`] = `
[
  "[90mâ”‚[39m
[32mâ—‡[39m  [0mtitle[0m [90mâ”€â”€â”€â•®[39m
[90mâ”‚[39m           [90mâ”‚[39m
[90mâ”‚[39m  [2mmessage[22m  [90mâ”‚[39m
[90mâ”‚[39m           [90mâ”‚[39m
[90mâ”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯[39m
",
]
`;
